src/Mod/Path/PathScripts/PathVcarve.py

```diff
@@ -0,0 +1,280 @@
+
+                # end points of candidate edge
+                cverts = [candidate.Edges[0].valueAt(candidate.Edges[0].FirstParameter),
+                    candidate.Edges[-1].valueAt(candidate.Edges[-1].LastParameter)]
+
+                # iterate the combination of endpoints. If a match is found,
+                # make an edge from the common endpoint to the other end of
+                # the candidate wire. Add the edge to the wire and return it.
+
+                # This generates a new edge rather than using the candidate to
+                # avoid vertexes with close but different vectors
+                for wvert in wireverts:
+                    for idx, cvert in enumerate(cverts):
+                        if PathGeom.pointsCoincide(wvert, cvert, obj.Tolerance):
+                            wireGrowing = True
+                            elist = mywire.Edges
+                            otherIndex = int(not(idx))
+                            newedge = Part.Edge(Part.Vertex(wvert),
+                                    Part.Vertex(cverts[otherIndex]))
+                            elist.append(newedge)
+                            mywire = Part.Wire(Part.__sortEdges__(elist))
+                            remaining.extend(unmatched[i+1:])
+                            return mywire, remaining, wireGrowing
+
+                # if not matched, add to remaining list to test later
+                remaining.append(candidate)
+
+            return mywire, remaining, wireGrowing
+
+        def getWires(candidateList):
+
+            chains = []
+            while len(candidateList) > 0:
+                cur_wire = Part.Wire(candidateList.pop(0))
+
+                wireGrowing = True
+                while wireGrowing:
+                    cur_wire, candidateList, wireGrowing = sortEm(cur_wire,
+                            candidateList)
+
+                chains.append(cur_wire)
+
+            return chains
+
+        def cutWire(w):
+            path = []
+            path.append(Path.Command("G0 Z{}".format(obj.SafeHeight.Value)))
+            e = w.Edges[0]
+            p = e.valueAt(e.FirstParameter)
+            path.append(Path.Command("G0 X{} Y{} Z{}".format(p.x, p.y,
+                obj.SafeHeight.Value)))
+            c = Path.Command("G1 X{} Y{} Z{} F{}".format(p.x, p.y, p.z,
+                obj.ToolController.HorizFeed.Value))
+            path.append(c)
+            for e in w.Edges:
+                path.extend(PathGeom.cmdsForEdge(e,
+                    hSpeed=obj.ToolController.HorizFeed.Value))
+
+            return path
+
+        pathlist = []
+        pathlist.append(Path.Command("(starting)"))
+        for f in Faces:
+            vd = Path.Voronoi()
+            insert_many_wires(vd, f.Wires)
+
+            vd.construct()
+
+            for e in vd.Edges:
+                e.Color = 0 if e.isPrimary() else 5
+            vd.colorExterior(1)
+            vd.colorExterior(4, lambda v: not f.isInside(v.toGeom(),
```

Comment:
need to set the correct z height otherwise all edges get filtered out:
`v.toGeom(f.BoundBox.ZMin)`
As a sanity check we could verify that the face actually lies in the XY plane - sooner or later somebody will want to engrave something non-flat.

Reply:
done